### PR TITLE
Add support for rounding to form expressions

### PR DESCRIPTION
**Pull request by martha**
## Description

Summary of changes:
- Add support for rounding functions to the form expression evaluator

[//]: # 'remove if not applicable'
Test against the json form here: https://qa-hmis.openpath.host/admin/forms/housing_needs_assessment

Screenshot 2025-08-04 at 3 57 20 PM

## Type of change
[//]: # (e.g., Bug fix, New feature, Documentation, Code clean-up, Dependency update)
New feature

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (eslint)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: src/modules/form/util/expressions/expressions.evaluateFormula.test.ts [hunk 1]
```diff
@@ -22,4 +22,30 @@ describe('evaluateFormula', () => {
       undefined
     );
   });
+
+  describe('rounding functions', () => {
```
Comment (gigxz) [inline src/modules/form/util/expressions/expressions.evaluateFormula.test.ts:57]: suggest adding tests where denominator value is nil and 0

**Review by gigxz — APPROVED** [review]
Looks good. I couldn't tell how all the edge cases were handled (missing and 0 denominators) so I suggested adding test cases for those.

File: src/modules/form/util/expressions/expressions.evaluateFormula.test.ts (hunk 1 above)
Comment (martha) [reply src/modules/form/util/expressions/expressions.evaluateFormula.test.ts:57]: Thanks! Sure, I added some tests for edge cases. I think these edge cases predate this PR, and I wasn't sure what the existing behavior was, so I investigated and found that our expression evaluator just does whatever the default javascript behavior is for edge cases like dividing by 0. With dividing by undefined, it will return undefined due to the try/catch here https://github.com/greenriver/hmis-frontend/blob/965c49be4cabae7fd5374385a6a0a40cf5e7925c/src/modules/form/util/expressions/formula.ts#L70-L79.

my understanding is that formula autofill is an advanced-user feature, and we expect users to ensure that the form behaves the way they want in terms of these edge cases. In this example for AMI/FPL,
- we are always dividing by an ami or fpl, which are not set to 0, so we should never divide by 0
- if ami or fpl is undefined, the autofill_when condition is not met, so we should never divide by undefined

Does that match your understanding?

Comment (gigxz) [reply src/modules/form/util/expressions/expressions.evaluateFormula.test.ts:57]: this seems ok, lets go ahead and merge @martha

**Comment by martha** [comment]
Sorry, I merged without checking, this should have been retargeted to release-176. I'll revert and retarget
